docs/document-scope.md

```diff
@@ -0,0 +1,73 @@
+# Access document scope
+
+CLUE tiles often need to access stuff at the document level to get their job done.
+
+Sometimes the code needing access is in the view layer (components and hooks) and sometimes it is in the model layer. The view layer has access to the models, but not vice versa, so anything described below about the model layer could be used by the views.
+
+Because the view can use the model layer mechanisms, code that is based on the model layer mechanisms can be used in more places.
+
+# Model layer
+
+## Traversing the tree
+Models can go up the MST tree to find the document or other parent and then use its views or actions or services.
+
+### Cases
+- `getDocumentContentFromNode` this looks explicitly at the string type of each parent node instead of using the MST "class" itself.
+- `getDocumentIdentifier` looks at the two kinds of parents of DocumentContent to construct an identifier.
+- `getTileModel` looks at the parent of of a tile content model to get the tile model
+
+### Benefits
+This is really simple to use. The calling code has full access to all of the properties, views, and actions of the parent. So new interface don't need to be created or modified, the calling code can just get what it needs.
+
+### Downsides
+This can make the code harder to test. A MST tree has to be created to test the model that is using it. If types are used to find the parent, a tree with those types has to be made, which can require lots of extra intermediate nodes.
+
+The approach of `getDocumentContentFromNode` means we can't use type "inheritance" to provide stripped down models which provide the necessary features. It would be possible to provide a mock model which has the name of DocumentContent to work around this, but that seems fragile.
+
```

Comment:
There's also the general dependency-inversion problem of children having explicit knowledge of their parents.